libsolidity/ast/Types.h

```diff
@@ -1247,6 +1247,38 @@ class FunctionType: public Type
 		/// Cannot be called.
 		Declaration,
 	};
+	struct Options
+	{
+		/// true if the function takes an arbitrary number of arguments of arbitrary types
+		bool arbitraryParameters = false;
+		/// true iff the gas value to be used is on the stack
+		bool gasSet = false;
+		/// true iff the value to be sent is on the stack
+		bool valueSet = false;
+		/// iff the salt value (for create2) to be used is on the stack
+		bool saltSet = false;
+		/// true iff the function is called as arg1.fun(arg2, ..., argn).
+		/// This is achieved through the "using for" directive.
+		bool bound = false;
+
+		static Options arbitraryParametersFactory()
+		{
+			Options result;
+			result.arbitraryParameters = true;
+			return result;
+		}
+		static Options ofFunction(FunctionType const& _type)
```

Comment:
After seeing it used, I think this would be slightly more readable:
```suggestion
		static Options fromFunctionType(FunctionType const& _type)
```